Server/SQL/schema.sql

```diff
@@ -0,0 +1,63 @@
+CREATE TABLE IF NOT EXISTS Card
+(
+    id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    deck_id VARCHAR(255) NOT NULL,
+    needs_review BOOLEAN NOT NULL DEFAULT FALSE,
+    front VARCHAR(255) NOT NULL,
+    back VARCHAR(255) NOT NULL,
+    position INTEGER NOT NULL
+
+    FOREIGN KEY (deck_id) REFERENCES Deck(uuid)
+);
+
+
+CREATE TABLE IF NOT EXISTS Deck
+(
+    uuid VARCHAR(255) NOT NULL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    owner VARCHAR(255),  -- theoretically could be null if a user deletes their account?
+    rating FLOAT DEFAULT NULL,
+    num_raitings INTEGER NOT NULL DEFAULT 0,
+    public BOOLEAN NOT NULL DEFAULT FALSE,
+    deleted BOOLEAN NOT NULL DEFAULT FALSE,
+    version INTEGER NOT NULL DEFAULT 0,
+    share_code VARCHAR(255) UNIQUE,
+
+    FOREIGN KEY (owner) REFERENCES Users(uuid)
+);
+
+
+CREATE TABLE IF NOT EXISTS Library
+(
+    user_id VARCHAR(255) NOT NULL,
+    deck_id VARCHAR(255) NOT NULL,
+
+    PRIMARY KEY (user_id, deck_id),
+    FOREIGN KEY (user_id) REFERENCES User(uuid),
+    FOREIGN KEY (deck_id) REFERENCES Deck(uuid)
+);
+
+
+CREATE TABLE IF NOT EXISTS Tag
+(
+    id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    tag VARCHAR(255) NOT NULL
+);
+
+
+CREATE TABLE IF NOT EXISTS DeckTag
+(
+    tag_id VARCHAR(255) NOT NULL,
+    deck_id VARCHAR (255) NOT NULL,
+
+    PRIMARY KEY (tag_id, deck_id),
+    FOREIGN KEY (tag_id) REFERENCES Tag(id),
+    FOREIGN KEY (deck_id) REFERENCES Deck(uuid)
+);
+
+
+CREATE TABLE IF NOT EXISTS User
```

Comment:
Since we're using Facebook authentication, we'll need to also store [the `userID` returned by the API](https://developers.facebook.com/docs/reference/ios/current/class/FBSDKAccessToken/#userID) so that we can identify our users. In any case, I think it's still a good idea to keep an additional UUID for each user should we decide to switch to a different authentication mechanism in the future.